streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -247,6 +277,19 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
         changelogReader.transitToRestoreActive();
     }
 
+    private void cleanUpTaskProducer(final Task task,
+                                     final Map<TaskId, RuntimeException> taskCloseExceptions) {
+        if (task.isActive()) {
+            try {
+                activeTaskCreator.closeAndRemoveTaskProducerIfNeeded(task.id());
+            } catch (final RuntimeException e) {
```

Comment:
We don't have unit test coverage for this exception case

Reply:
Added test `shouldThrowWhenHandlingClosingTasksOnProducerCloseError`